AmplifyPlugins/Auth/AWSAuthPlugin/Support/Utils/AuthCognitoSignedOutSessionHelper.swift

```diff
@@ -0,0 +1,161 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSMobileClient
+import AWSPluginsCore
+
+struct AuthCognitoSignedOutSessionHelper {
+
+    /// Creates a signedOut session information with valid identityId and aws credentials.
+    /// - Parameters:
+    ///   - identityId: Valid identity id for the current signedOut session
+    ///   - awsCredentials: Valid AWS Credentials for the current signedOut session
+    /// - Returns: Returns a valid signedOut session
+    static func signedOutSession(identityId: String,
+                                 awsCredentials: AuthAWSCredentials) -> AWSAuthCognitoSession {
+        let tokensError = cognitoTokensSignedOutError()
+        let userSubError = userSubSignedOutError()
+        let authSession = AWSAuthCognitoSession(isSignedIn: false,
+                                                userSubResult: .failure(userSubError),
+                                                identityIdResult: .success(identityId),
+                                                awsCredentialsResult: .success(awsCredentials),
+                                                cognitoTokensResult: .failure(tokensError))
+        return authSession
+    }
+
+    /// Creates a signedOut session by handling the error
+    /// - Parameter error: Error received while fetching session information
+    /// - Returns: SignedOut session with isSignedIn = false and appropriate error result for other values.
+    static func signedOutSession(withError error: Error) -> AWSAuthCognitoSession {
+
+        if let urlError = error as NSError?, urlError.domain == NSURLErrorDomain {
+            return offlineSignedOutSession()
```

Comment:
nit: here and elsewhere: name these factory methods with `make` to clarify that they're creating something, not doing some side effect. If I'm mistaken and the methods are also changing system state, then reflect that in the name too, or break the side effect into a different method